tasks/db_deploy/migrations/migrate_versions_4_to_5/migrate_sql.py

```diff
@@ -0,0 +1,311 @@
+        """
+    )
+
+
+def reconcile_s3_object_table_sql() -> TextClause:
+    """
+    Full SQL for creating the reconcile_s3_object table. 
+
+    Returns:
+        (sqlalchemy.sql.element.TextClause): SQL for creating reconcile_s3_object table.
+    """
+    return text(
+        """
+            -- Create reconcile_s3_object table
+            CREATE TABLE IF NOT EXISTS reconcile_s3_object  ( 
+              job_id               	int8 NOT NULL
+            , orca_archive_location	text NOT NULL
+            , key_path             	text NOT NULL
+            , etag                 	text NOT NULL
+            , last_update          	timestamp with time zone NOT NULL
+            , size_in_bytes        	int8 NOT NULL
+            , storage_class        	text NOT NULL
+            , delete_marker        	bool NOT NULL
+            , CONSTRAINT FK_reconcile_job_s3_objects FOREIGN KEY(job_id) REFERENCES reconcile_job(id)
+            , CONSTRAINT PK_reconcile_s3_object PRIMARY KEY(orca_archive_location, key_path)
+                ) 
+            PARTITION BY LIST (orca_archive_location);
+
+            -- Comment  
+            COMMENT ON TABLE reconcile_s3_object IS 'Temporary table that holds the listing from the ORCA S3 bucket to use for comparisons against the ORCA catalog.';
+            COMMENT ON COLUMN reconcile_s3_object.job_id IS 'Job the S3 listing is a part of for the comparison. Foreign key to the reconcile jobs table.';
+            COMMENT ON COLUMN reconcile_s3_object.orca_archive_location IS 'ORCA S3 Glacier bucket name where the file is stored.';
+            COMMENT ON COLUMN reconcile_s3_object.key_path IS 'Full path and file name of the object in the S3 bucket.';
+            COMMENT ON COLUMN reconcile_s3_object.etag IS 'AWS etag value from the s3 inventory report.';
+            COMMENT ON COLUMN reconcile_s3_object.last_update IS 'AWS Last Update from the s3 inventory report.';
+            COMMENT ON COLUMN reconcile_s3_object.size_in_bytes IS 'AWS size of the file in bytes from the s3 inventory report.';
+            COMMENT ON COLUMN reconcile_s3_object.storage_class IS 'AWS storage class the object is in from the s3 inventory report.';
+            COMMENT ON COLUMN reconcile_s3_object.delete_marker IS 'Set to `True` if object is a delete marker.';
+        """
+    )
+
+def reconcile_s3_object_partition_sql(partition_name: str) -> TextClause:
+    """
+    Full SQL for creating the reconcile_s3_object partition table. 
+
+    Returns:
+        (sqlalchemy.sql.element.TextClause): SQL for creating reconcile_s3_object partition table.
+    """
+    if partition_name is None or len(partition_name) == 0:
+        raise Exception("partition name is not set properly")
+    else:
+        return text(
+            f"""
+                -- Create orca_archive_location_:bucket_name
+                CREATE TABLE {partition_name}
+                (
+                PARTITION OF reconcile_s3_object
```

Comment:
Extra stuff for adding constraints:
https://github.com/nasa/cumulus-orca/pull/241#discussion_r805021450

Reply:
Updated per previous comments.